ext/TensorKitChainRulesCoreExt.jl

```diff
@@ -111,19 +121,17 @@ function ChainRulesCore.rrule(::typeof(⊗), A::AbstractTensorMap, B::AbstractTe
         dA_ = @thunk begin
             ipA = (codomainind(A), domainind(A))
             pB = (allind(B), ())
-            dA = zerovector(A,
-                            TensorOperations.promote_contract(scalartype(ΔC),
-                                                              scalartype(B)))
-            dA = tensorcontract!(dA, ipA, ΔC, pΔC, :N, B, pB, :C)
+            dA = zerovector(A, promote_contract(scalartype(ΔC), scalartype(B)))
+            tB = twist(B, filter(x -> isdual(space(B, x)), allind(B)))
+            dA = tensorcontract!(dA, ipA, ΔC, pΔC, :N, tB, pB, :C)
```

Comment:
I am wondering whether it would not be better to write these rules using `planarcontract!`. The tensor product method `⊗` works for any two tensors of the same spacetype.

Reply:
I agree, but currently `planarcontract!` does not have support for conj flags (which we could add manually, but that feels a bit annoying since we should have this in the near future anyways), and I think there are some assumptions on `pAB` which we'd have to explicitly check here. Maybe a TODO comment for after the planar updates would be better?

Reply:
I agree that a TODO will do for now. But maybe using the (hermitian) adjoint of `A` and of `B` in the definition for the (pullback) adjoints of `B` and and `A` would be very natural.